activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -96,7 +96,7 @@ def self.build_mem_cache(*addresses) # :nodoc:
         if pool_options.empty?
           Dalli::Client.new(addresses, options)
         else
-          ConnectionPool.new(pool_options) { Dalli::Client.new(addresses, options.merge(threadsafe: false)) }
+          ConnectionPool.new(pool_options) { Dalli::Client.new(addresses, options.reverse_merge(threadsafe: false)) }
```

Comment:
hmm, I see the same introduction of this in https://github.com/petergoldstein/dalli/pull/828#discussion_r1117111414 without comment